nixos/modules/services/mail/rss2email.nix

```diff
@@ -0,0 +1,226 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.rss2email;
+  conf = pkgs.writeText "rss2email.cfg" (lib.generators.toINI {} ({
+      DEFAULT = default-config // cfg.configuration // { to = cfg.to; };
+    } // lib.mapAttrs' (name: feed: nameValuePair "feed.${name}" (
+      { inherit (feed) url; } //
+      lib.optionalAttrs (feed.to != null) { inherit (feed) to; }
+    )) cfg.feeds
+  ));
+
+  default-config = {
+    use-8bit = "False";
+    force-from = "False";
+    use-publisher-email = "False";
+    name-format = "{feed-title}: {author}";
+    proxy = "";
+    feed-timeout = "60";
+    active = "True";
+    digest = "False";
+    date-header = "False";
+    date-header-order = "modified, issued, created, expired";
+    bonus-header = "";
+    trust-guid = "True";
+    trust-link = "False";
+    encodings = "US-ASCII, ISO-8859-1, UTF-8, BIG5, ISO-2022-JP";
+    post-process = "";
+    digest-post-process = "";
+    html-mail = "False";
+    use-css = "False";
+    css = builtins.replaceStrings ["\n"] [""] ''
+      h1 {
+        font: 18pt Georgia, "Times New Roman";
+      }
+      body {
+        font: 12pt Arial;
+      }
+      a:link {
+        font: 12pt Arial;
+        font-weight: bold;
+        color: #0000cc;
+      }
+      blockquote {
+        font-family: monospace;
+      }
+      .header {
+        background: #e0ecff;
+        border-bottom: solid 4px #c3d9ff;
+        padding: 5px;
+        margin-top: 0px;
+        color: red;
+      }
+      .header a {
+        font-size: 20px;
+        text-decoration: none;
+      }
+      .footer {
+        background: #c3d9ff;
+        border-top: solid 4px #c3d9ff;
+        padding: 5px;
+        margin-bottom: 0px;
+      }
+      #entry {
+        border: solid 4px #c3d9ff;
+      }
+      #body {
+        margin-left: 5px;
+        margin-right: 5px;
+      }
+    '';
+    unicode-snob = "False";
+    links-after-each-paragraph = "False";
+    body-width = "0";
+    email-protocol = "sendmail";
+    sendmail = "sendmail";
+    smtp-auth = "False";
+    smtp-username = "username";
+    smtp-password = "password";
+    smtp-server = "smtp.yourisp.net:25";
+    smtp-ssl = "False";
+    smtp-ssl-protocol = "SSLv3";
+    imap-auth = "False";
+    imap-username = "username";
+    imap-password = "password";
+    imap-server = "imap.yourisp.net";
+    imap-port = "143";
+    imap-ssl = "False";
+    imap-mailbox = "INBOX";
+    verbose = "warning";
+  };
+
+in {
+
+  ###### interface
+
+  options = {
+
+    services.rss2email = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable rss2email.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.rss2email;
+        defaultText = "pkgs.rss2email";
+        description = "The rss2email package to use.";
+      };
```

Comment:
Is there a need to have this configurable? Is there an alternate package? If not, it's best to not provide this option because nobody will use it. And in any case, overlays can still change the value used by this module.

Reply:
There is no alternate package for the time being. However, I know I would personally use an alternate package (unreleased version), and I went on the `services.opensmtpd.package` precedent.

Also, I personally think that overlays are not really pleasant to use, and NUR packages are a much cleaner way of handling package changes, but a NUR package would require a local overlay to set it up with this module if the option was absent, so…

Reply:
I'd rather introduce this option when an actual need for it arises then. Upon a quick count, there's about 600 service modules in NixOS, and about 120 of them have a `package` option. If all of them would also introduce such an option, this would slow down evaluation quite a bit.

The 3 uses I see for a package option are:
- There are alternate packages you would want to use and being able to use different packages is an integral part of the module. Having an option makes this possibility discoverable. `postgresql` is such an example
- Using an overlay for it would require a lot of rebuilds, because overlays not only change the package the service uses, but all the other parts of the system as well. `curl` would be such an example (don't know of an example for a service though).
- Using an overlay for it would change some other part of the system you don't want to change. E.g. 2 services that need the same package but different versions of it. An overlay couldn't change just one of them.

Reply:
Well… I'm not of the same opinion, thinking that 600 options is basically nothing compared to the one-option-per-upstream-option vs. one-option-per-upstream-file, but let's agree to disagree :) I'm removing this option, as the usual behavior in NixOS appears to be not to have the `package` option.